test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -889,49 +872,11 @@ def _validate_docs(self):
         if self.object_name.startswith('_') and not os.path.islink(self.object_path):
             filename_deprecated_or_removed = True
 
-        # Have to check the metadata first so that we know if the module is removed or deprecated
-        metadata = None
-        if not self.collection:
-            if not bool(doc_info['ANSIBLE_METADATA']['value']):
-                self.reporter.error(
-                    path=self.object_path,
-                    code='missing-metadata',
-                    msg='No ANSIBLE_METADATA provided'
-                )
-            else:
-                if isinstance(doc_info['ANSIBLE_METADATA']['value'], ast.Dict):
-                    metadata = ast.literal_eval(
-                        doc_info['ANSIBLE_METADATA']['value']
-                    )
-                else:
-                    self.reporter.error(
-                        path=self.object_path,
-                        code='missing-metadata-format',
-                        msg='ANSIBLE_METADATA was not provided as a dict, YAML not supported'
-                    )
-
-            if metadata:
-                self._validate_docs_schema(metadata, metadata_1_1_schema(),
-                                           'ANSIBLE_METADATA', 'invalid-metadata-type')
-                # We could validate these via the schema if we knew what the values are ahead of
-                # time.  We can figure that out for deprecated but we can't for removed.  Only the
-                # metadata has that information.
-                if 'removed' in metadata['status']:
-                    removed = True
-                if 'deprecated' in metadata['status']:
-                    deprecated = True
-                if (deprecated or removed) and len(metadata['status']) > 1:
-                    self.reporter.error(
-                        path=self.object_path,
-                        code='missing-metadata-status',
-                        msg='ANSIBLE_METADATA.status must be exactly one of "deprecated" or "removed"'
-                    )
-        else:
-            # We are testing a collection
-            if self.routing and self.routing.get('plugin_routing', {}).get('modules', {}).get(self.name, {}).get('deprecation', {}):
-                # meta/routing.yml says this is deprecated
-                routing_says_deprecated = True
-                deprecated = True
+        # We are testing a collection
+        if self.routing and self.routing.get('plugin_routing', {}).get('modules', {}).get(self.name, {}).get('deprecation', {}):
```

Comment:
Should this be inside a `if self.collection:`?

Reply:
unsure, since routing can possibly handle base deprecations also